Remotion/Web/IntegrationTests/SmartPage/HistoryBackHandlingTests.cs

```diff
@@ -0,0 +1,159 @@
+      var sessionStorageEntryCountDifference = sessionStorageEntryCount - initialSessionStorageEntryCount;
+      Assert.That(sessionStorageEntryCountDifference, Is.EqualTo(2));
+      Assert.That(sessionStorageEntryKeyValue, Is.EqualTo((sessionStorageEntryCount - 1).ToString()));
+    }
+
+    [Test]
+    public void NavigateToSamePage_IncreasesSessionStorageEntries ()
+    {
+      var initialPage = StartWithCleanSessionStorage("WebTabStripTest.wxe");
+      var jsExecutor = JavaScriptExecutor.GetJavaScriptExecutor(initialPage.Scope);
+      var initialSessionStorageEntryCount = GetSessionStorageEntryCount(jsExecutor);
+
+      Start<WxePageObject>("WebTabStripTest.wxe");
+
+      var sessionStorageEntryCount = GetSessionStorageEntryCount(jsExecutor);
+      var sessionStorageEntryKeyValue = GetSessionStorageEntry(jsExecutor, "pageTokenEntryCount");
+      var sessionStorageEntryCountDifference = sessionStorageEntryCount - initialSessionStorageEntryCount;
+      Assert.That(sessionStorageEntryCountDifference, Is.EqualTo(1));
+      Assert.That(sessionStorageEntryKeyValue, Is.EqualTo((sessionStorageEntryCount - 1).ToString()));
+    }
+
+    [Test]
+    public void NavigateToDifferentPage_IncreasesSessionStorageEntries ()
+    {
+      var initialPage = StartWithCleanSessionStorage("WebButtonTest.wxe");
+      var jsExecutor = JavaScriptExecutor.GetJavaScriptExecutor(initialPage.Scope);
+      var initialSessionStorageEntryCount = GetSessionStorageEntryCount(jsExecutor);
+
+      Start<WxePageObject>("WebTabStripTest.wxe");
+
+      var sessionStorageEntryCount = GetSessionStorageEntryCount(jsExecutor);
+      var sessionStorageEntryKeyValue = GetSessionStorageEntry(jsExecutor, "pageTokenEntryCount");
+      var sessionStorageEntryCountDifference = sessionStorageEntryCount - initialSessionStorageEntryCount;
+      Assert.That(sessionStorageEntryCountDifference, Is.EqualTo(1));
+      Assert.That(sessionStorageEntryKeyValue, Is.EqualTo((sessionStorageEntryCount - 1).ToString()));
+    }
+
+    [Test]
+    public void NewSessionStorageEntry_OverThreshold_ClearsOldEntries ()
+    {
+      var initialPage = StartWithCleanSessionStorage("WebButtonTest.wxe");
+      var jsExecutor = JavaScriptExecutor.GetJavaScriptExecutor(initialPage.Scope);
+
+      foreach (var _ in Enumerable.Range(0, 149))
+        Start<WxePageObject>("WebTabStripTest.wxe");
+
+      Assert.That(GetSessionStorageEntryCount(jsExecutor), Is.EqualTo(151));
+      Assert.That(GetSessionStorageEntry(jsExecutor, "pageTokenEntryCount"), Is.EqualTo(150.ToString()));
+
+      Start<WxePageObject>("WebTabStripTest.wxe");
+
+      var sessionStorageEntryCount = GetSessionStorageEntryCount(jsExecutor);
+      var sessionStorageEntryKeyValue = GetSessionStorageEntry(jsExecutor, "pageTokenEntryCount");
+      Assert.That(sessionStorageEntryCount, Is.EqualTo(101));
+      Assert.That(sessionStorageEntryKeyValue, Is.EqualTo((sessionStorageEntryCount - 1).ToString()));
```

Comment:
For a complete test, we should always include a check that the added value in the sessionstate is actually correct. For the Threshold-test, test that the most recent entry is still there, plus probably the 3-from-last or so, to document the correct sorting order.

The other tests are probably already covered via the status-message tests, but the Threshold-test isn't. It would be okay to just check for expected page token values (read from the hidden field after loading the respective page load and then used for the assertion.

Reply:
Please have a look at [`4400fc9` (#332)](https://github.com/re-motion/Framework/pull/332/commits/4400fc9ca662f30c0a132d8f2e9ab53e29ad423a).